### PR TITLE
Log server map send cycle time.

### PR DESCRIPTION
**Pull request by lhofhansl**
This just adds an infostream line about how much time the server took to update/send the complete map to the client.

For server admins that can provide a good top bound about the best value for `server_unload_unused_data_timeout`.
In the future we can use that knowledge to adapt that unload timeout.

## To do

This PR is Ready for Review.

## How to test

Start MT with --info, watch the "Time to send complete map:" messages.
Note how for large viewing_ranges this can be in the minutes.

### CODE REVIEW

File: src/clientiface.cpp [hunk 1]
```diff
@@ -384,6 +385,8 @@ void RemoteClient::GetNextBlocks (
 		if (d > full_d_max) {
 			new_nearest_unsent_d = 0;
 			m_nothing_to_send_pause_timer = 2.0f;
+			infostream << "Server: Time to send complete map: " << m_map_send_completion_timer << "s" << std::endl;
+			m_map_send_completion_timer = 0.0f;
```
Comment (sfan5) [inline src/clientiface.cpp:389]: this needs to be per-client somehow
also maybe this better fits into the profiler?

Comment (lhofhansl) [reply src/clientiface.cpp:389]: Hmm... Good point. EIther log the client or send to the profiler.

Comment (lhofhansl) [reply src/clientiface.cpp:389]: I think profiler is actually not a good idea, as this value would not be available frequently. I'll log the player name instead... It's really just an aid for a server admin to set the unload timeout correctly - or for us DEVs to see what is going on.

Comment (lhofhansl) [reply src/clientiface.cpp:389]: We also do not log the client for other map related changes, so that might be confusing.

**Comment by lhofhansl** [comment]
How about this @sfan5 ?

File: src/clientiface.cpp [hunk 2]
```diff
@@ -384,6 +385,8 @@ void RemoteClient::GetNextBlocks (
 		if (d > full_d_max) {
 			new_nearest_unsent_d = 0;
 			m_nothing_to_send_pause_timer = 2.0f;
+			infostream << "Server: map update for player " << m_name << " (peer " << peer_id << "), time: " << m_map_send_completion_timer << "s" << std::endl;
```
Comment (sfan5) [inline src/clientiface.cpp:388]: apparently we use `o<<"RemoteClient "<<peer_id<<": "` in some other case so that might be better, pick one

also I'd reword it to "full map send completed after Xs, restarting"

**Review by sfan5 — APPROVED** [review]

**Comment by lhofhansl** [comment]
Changed the message. This is what I plan to commit.
